mutablesecurity/solutions/implementations/duplicati/code.py

```diff
@@ -0,0 +1,693 @@
+            src=template_path,
+            dest="/opt/mutablesecurity/duplicati/duplicati.conf",
+            configuration=j2_values,
+            name="Copy the generated configuration into duplicati's folder.",
+        )
+
+    @staticmethod
+    @deploy
+    class ExcludeFilesValue(FactBase):
+        command = "echo 'hi'"
+
+        @staticmethod
+        def process(output: typing.List[str]) -> str:
+            return (
+                "cat /opt/mutablesecurity/duplicati/duplicati.conf | grep"
+                f" {IDENTIFIER} | cut -d : -f 2"
+            )
+
+    IDENTIFIER = "exclude_files_attributes"
+    DESCRIPTION = "Don't backup files which have this attribute"
+    INFO_TYPE = StringDataType
+    PROPERTIES = [
+        InformationProperties.CONFIGURATION,
+        InformationProperties.OPTIONAL,
+        InformationProperties.WITH_DEFAULT_VALUE,
+        InformationProperties.NON_DEDUCTIBLE,
+        InformationProperties.WRITABLE,
+    ]
+    DEFAULT_VALUE = "Temporary"
+    GETTER = ExcludeFilesValue
+    SETTER = set_configuration
+
+
+class Passphrase(BaseInformation):
+    @staticmethod
+    @deploy
+    def set_configuration(
+        old_value: typing.Any, new_value: typing.Any
+    ) -> None:
+        template_path = os.path.join(
+            os.path.dirname(__file__), "files/duplicati.conf.j2"
+        )
+        j2_values = {"passphrase": Passphrase.get()}
+        files.template(
+            src=template_path,
+            dest="/opt/mutablesecurity/duplicati/duplicati.conf",
+            configuration=j2_values,
+            name="Copy the generated configuration into duplicati's folder.",
+        )
+
+    @staticmethod
+    @deploy
+    class PassphraseValue(FactBase):
+        command = "echo 'hi'"
+
+        @staticmethod
+        def process(output: typing.List[str]) -> str:
+            return (
+                "cat /opt/mutablesecurity/duplicati/duplicati.conf | grep"
+                f" {IDENTIFIER} | cut -d : -f 2"
+            )
+
+    IDENTIFIER = "passphrase"
+    DESCRIPTION = "This value represents the value by encryption key"
+    INFO_TYPE = StringDataType
+    PROPERTIES = [
+        InformationProperties.CONFIGURATION,
+        InformationProperties.OPTIONAL,
+        InformationProperties.WITH_DEFAULT_VALUE,
+        InformationProperties.NON_DEDUCTIBLE,
+        InformationProperties.WRITABLE,
+    ]
+    DEFAULT_VALUE = "mutablesecutiy"
+    GETTER = PassphraseValue
+    SETTER = set_configuration
+
+
+class RestorePath(BaseInformation):
+    IDENTIFIER = "restore_path"
+    DESCRIPTION = "Path of directory you want to restore your backup"
+    INFO_TYPE = StringDataType
+    PROPERTIES = [
+        InformationProperties.CONFIGURATION,
+        InformationProperties.OPTIONAL,
+        InformationProperties.WITH_DEFAULT_VALUE,
+        InformationProperties.NON_DEDUCTIBLE,
+        InformationProperties.WRITABLE,
+    ]
+    DEFAULT_VALUE = "/opt/mutablesecurity/duplicati/restorePath"
```

Comment:
Just wondering why the restore path should have a default value. What's the reasoning behind this choice?